### PR TITLE
Added GitHub pull request template

### LINKED ISSUE #1307: Create Issue and Pull Request template
The issue template should ask for basic information such as:
* rook version currently running
* Kubernetes version
* Ask for logs (show the commands in html comments)
* OS (CoreOS, Atomic, Ubuntu, etc)
* Kernel version `uname -a`
* Ceph status `ceph -s` + `ceph mon_status`
* Other infos?

The pull request template should contain information such as:
* Link to contributing/development flow doc
* Others?

***

**ToDo**:
- [x] Issue template
- [ ] Pull Request

### PR DESCRIPTION
**Pull request by galexrt**
This is the second part of #1307.

The rendered version of the template can be found here: https://github.com/galexrt/rook/blob/feature/1307/.github/PULL_REQUEST_TEMPLATE.md

[skip ci]

***

Resolves #1307.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+* Feature
+
+Description of your changes:
+
+Which issue is resolved by this Pull Request:
+
+Does your change break an existing feature:
+
+Checklist:
+- [ ] Read the [contribution guide](https://github.com/rook/rook/blob/master/CONTRIBUTING.md).
```
Comment (travisn) [inline .github/PULL_REQUEST_TEMPLATE.md]: I don't think the first two bullets are necessary. They are covered sufficiently by the comment on line 1. What about the following checklist?
- Documentation has been updated, if necessary
- PendingReleaseNotes.md has been updated with breaking and/or notable changes, if necessary

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<!-- Please take a look at our [Contributing](https://rook.io/docs/rook/master/development-flow.html)
+documentation before submitting a Pull Request!
+Thank you for contributing to Rook! -->
+
+Is this a fix for an issue or a feature implementation?
```
Comment (travisn) [inline .github/PULL_REQUEST_TEMPLATE.md]: The question of fix/feature should be sufficiently answered by the description, as well as the link to the issue that is fixed. Can we remove lines 5-8?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+<!-- Remove only one -->
+* Fix
+* Feature
+
+Description of your changes:
+
+Which issue is resolved by this Pull Request:
```
Comment (travisn) [inline .github/PULL_REQUEST_TEMPLATE.md]: We should include a [keyword](https://help.github.com/articles/closing-issues-using-keywords/) here such that github will automatically resolve the issue. For example:
`Fixes: #<issueNumber>`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+Does your change break an existing feature:
```
Comment (travisn) [inline .github/PULL_REQUEST_TEMPLATE.md]: This should be part of the description above (implicit), and will be covered by the checklist below. So can we remove this separate line?

**Comment by galexrt** [comment]
@travisn I added your feedback.

**Pull request merged**